### ISSUE TITLE
Feature Request: data source for aws_iam_role

### ISSUE THREAD

**Issue by iancward** [hashicorp/terraform#12825]
Hi there,
It is currently possible to create a data source for a number of different AWS resources; however, it looks like aws_iam_role is missing. My team has a repository for all of our baseline infrastructure templates, and we put templates for applications in their own, separate, repository. This is intended to reduce risk for big changes. As such, we'd like to be able to define aws_iam_roles (e.g. for monitoring roles) in our central repository, and then have the application templates refer to it via a data definition. 

Resources such as aws_db_cluster_instance require the ARN of the IAM role for monitoring, and we'd rather not have to hard-code it in these separate resources.

### Terraform Version
0.9.0

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_iam_role

### Terraform Configuration Files
```hcl
resource "aws_iam_role" "my-awesome-iam-role" {
  name = "my-awesome-iam-role"
  assume_role_policy = "${data.aws_iam_policy_ducument.my-awesome-iam-role-assume-policy}"
}
```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
I'd like to be able to add a data source that is a reference to a pre-existing IAM role.

### Actual Behavior
I can currently not use a data source to reference a pre-existing IAM role. I have to either hard-code the ARN or use the external data feature to look up the ARN via the AWS CLI.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
+1 for the data sources feature. I'm looking at terraform for the first time in a while, and they're a welcome addition.

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? 
Not that I could find.

**Comment by mathematician** [comment]
I am about to submit a pull request for this

**Comment by stack72** [comment]
Closed via #13213
